### PR TITLE
Remove team and user from team, fix team edit

### PR DESCRIPTION
**Pull request by YrPryhoda**
* Only team lead can edit team info, add users to team, remove team,
* Each of users which consists in team can add, edit, remove links

### CODE REVIEW

File: server/src/repositories/teams.repository.ts [hunk 1]
```diff
@@ -105,4 +105,19 @@ export class TeamRepository extends Repository<Team> {
...
+			.getOne();
+
+		if (!team) {
+			throw new Error('Team was not found');
+		}
+
+		team.users = team.users?.filter((user: any) => user.id !== userId);
+
+		return this.save(team);
+	}
```
Comment (R-Mielamud) [inline server/src/repositories/teams.repository.ts:122]: this.update

File: server/src/repositories/teams.repository.ts [hunk 2]
```diff
@@ -105,4 +105,19 @@ export class TeamRepository extends Repository<Team> {
 
 		return this.save(team);
 	}
+
+	async removeUserFromTeam(userId: string, teamId: string) {
+		const team = await this.createQueryBuilder('Team')
+			.where('Team.id = :id', { id: teamId })
+			.leftJoinAndSelect('Team.users', 'Users')
+			.getOne();
```
Comment (R-Mielamud) [inline server/src/repositories/teams.repository.ts:113]: this.findOne

**Review by R-Mielamud — CHANGES_REQUESTED** [review]

File: server/src/repositories/teams.repository.ts (hunk 1 above)
Comment (YrPryhoda) [reply server/src/repositories/teams.repository.ts:122]: this.update doesn't return saved item, but i need to get results back on frontend part

File: client/src/components/TeamDevsCard/index.tsx [hunk 3]
```diff
@@ -19,9 +22,12 @@ const TeamDevsCard = ({
 	showAddPeopleModal,
 	teamOwner,
 	changeMainFields,
-	description = '',
+	confirmDelete,
+	description = ' ',
 	name = ' ',
 }: CardProps) => {
```
[the remark this answers is not in the document]
Comment (YrPryhoda) [reply client/src/components/TeamDevsCard/index.tsx:28]: ok

File: client/src/components/TeamsMembersCard/index.tsx [hunk 4]
```diff
@@ -22,21 +26,45 @@ export interface User {
...
+	};
+
+	const removeUser = (id: string) => {
+		removeUserFromTeam(id);
+		setViewUser(undefined);
+		setRemoveUserModal(false);
+	};
+
+	const showDeleteUserIcon = (el: User) =>
+		teamOwner && authUser?.id === teamOwner.id && viewUser?.id === el.id && viewUser?.id !== teamOwner.id;
```
[the remark this answers is not in the document]
Comment (YrPryhoda) [reply client/src/components/TeamsMembersCard/index.tsx:68]: fixed

File: client/src/components/TeamsMembersCard/index.tsx [hunk 5]
```diff
@@ -66,7 +94,21 @@ const TeamsMembersCard = ({ title, removeFromTeam, teammates = [] }: Props) => {
 							<div className={styles.user_info}>
 								<p> {fullUserName(el.firstName, el.lastName)}</p>
 								<p className={styles.metainfo}>{el.jobTitle}</p>
+								{showDeleteUserIcon(el) && (
+									<Icon
+										name="delete"
+										onClick={() => onDeleteUserClick(el)}
+										className={styles.user_delete_btn}
+									/>
```
[the remark this answers is not in the document]
Comment (YrPryhoda) [reply client/src/components/TeamsMembersCard/index.tsx:102]: renamed

File: client/src/components/TeamsMembersCard/index.tsx [hunk 6]
```diff
@@ -66,7 +94,21 @@ const TeamsMembersCard = ({ title, removeFromTeam, teammates = [] }: Props) => {
...
+							{removeUserModal && (
+								<RemoveUserModal
+									user={userToRemove}
+									confirm={removeUser}
+									setShowDelete={setRemoveUserModal}
+								/>
+							)}
 							{modal && viewUser?.id === el.id && (
 								<ModalViewProfile key={el.id} user={viewUser} onClose={hideModal} />
 							)}
```
[the remark this answers is not in the document]
Comment (YrPryhoda) [reply client/src/components/TeamsMembersCard/index.tsx:114]: ok

File: client/src/components/TeamsMembersCard/removeUserModal.tsx [hunk 7]
```diff
@@ -0,0 +1,41 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (YrPryhoda) [reply client/src/components/TeamsMembersCard/removeUserModal.tsx:2]: renamed component file

File: client/src/containers/TeamPage/logic/saga.ts [hunk 8]
```diff
@@ -7,12 +7,16 @@ import {
 	getTeamsUsers,
 	findUsersColleagues,
 	addUsersToTeam,
+	deleteTeamRequest,
+	removeUserFromTeamRequest
 } from 'services/team.service';
 import { all, put, takeEvery, call } from 'redux-saga/effects';
```
[the remark this answers is not in the document]
Comment (YrPryhoda) [reply client/src/containers/TeamPage/logic/saga.ts:13]: ok

File: client/src/containers/TeamPage/logic/saga.ts [hunk 9]
```diff
@@ -89,6 +98,27 @@ function* inviteUsersToTeam(props: Props) {
...
+}
+
+function* removeUserFromTeam(props: RemoveUserFromTeam) {
+	try {
+		const team = yield call(removeUserFromTeamRequest, props.userId, props.teamId);
+		yield put(actions.deletePeopleFromTeamSuccess({ users: team.users }));
+		NotificationManager.info('User has been removed', 'Info', 4000);
+	} catch (error) {
+		NotificationManager.error('Error delete team', 'Error', 4000);
+	}
```
[the remark this answers is not in the document]
Comment (YrPryhoda) [reply client/src/containers/TeamPage/logic/saga.ts:119]: good idea

File: server/src/repositories/teams.repository.ts (hunk 2 above)
Comment (YrPryhoda) [reply server/src/repositories/teams.repository.ts:113]: I suppose it is not required, but just a matter of opinion )

File: server/src/repositories/teams.repository.ts (hunk 1 above)
Comment (R-Mielamud) [reply server/src/repositories/teams.repository.ts:122]: ok

File: server/src/repositories/teams.repository.ts (hunk 2 above)
Comment (R-Mielamud) [reply server/src/repositories/teams.repository.ts:113]: this.findOne is cleaner

File: client/src/containers/TeamPage/logic/reducer.ts [hunk 10]
```diff
@@ -92,6 +92,9 @@ export const teamReducer = createReducer<TeamState>(initialState, {
 			},
 		};
 	},
+	[actionTypes.DELETE_TEAM_SUCCESS](state: TeamState) {
+		return initialState
+	},
 
 	[actionTypes.LOADING](state: TeamState) {
```
[the remark this answers is not in the document]
Comment (YrPryhoda) [reply client/src/containers/TeamPage/logic/reducer.ts]: renamed

File: client/src/helpers/history.helper.ts [hunk 11]
```diff
@@ -1,3 +1,3 @@
 import { createBrowserHistory } from 'history';
 
-export default createBrowserHistory();
+export default createBrowserHistory();
```
Comment (R-Mielamud) [inline client/src/helpers/history.helper.ts:3]: Add newline after createBrowserHistory(); (end-of-source line)

Comment (YrPryhoda) [reply client/src/helpers/history.helper.ts]: done

**Review by R-Mielamud — APPROVED** [review]

**Pull request merged**